railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -592,6 +605,35 @@ def after_bundle(&block) # :doc:
       def get_builder_class
         defined?(::AppBuilder) ? ::AppBuilder : Rails::AppBuilder
       end
+
+      def devcontainer_dependencies
+        return @devcontainer_dependencies if @devcontainer_dependencies
+
+        @devcontainer_dependencies = []
+
+        @devcontainer_dependencies << "selenium" if depends_on_system_test?
+        @devcontainer_dependencies << "redis" unless options.skip_action_cable? && options.skip_active_job?
+        @devcontainer_dependencies << "postgres" if options.database == "postgresql"
+        @devcontainer_dependencies << "mysql" if options.database == "mysql"
+        @devcontainer_dependencies << "mariadb" if options.database == "trilogy"
+
+        @devcontainer_dependencies
+      end
+
+      def devcontainer_variables
+        return @devcontainer_variables if @devcontainer_variables
+
+        @devcontainer_variables = {}
+
+        @devcontainer_variables["CAPYBARA_SERVER_PORT"] = "45678" if depends_on_system_test?
+        @devcontainer_variables["SELENIUM_HOST"] = "selenium" if depends_on_system_test?
+        @devcontainer_variables["REDIS_URL"] = "redis://redis:6379/1" unless options.skip_active_job? && options.skip_action_cable?
+        @devcontainer_variables["DB_HOST"] = "postgres" if options.database == "postgresql"
```

Comment:
Why don't we use `DATABASE_URL`?

Reply:
Trilogy and mysql2 templates use `host` already, so we will continue with that and use `host` in the postgres template to remain consistent.